ui/lib/kv/addon/components/kv-patch/json-form.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module KvPatchJsonForm
+ * @description
+ * This component renders one of two ways to patch a KV v2 secret (the other is using the KvPatch::Editor::Form).
+ * This input instantiates empty so it does not receive @subkeys as an argument.
+ *
+ * @example
+ * <KvPatch::JsonForm @onSubmit={{perform this.save}} @onCancel={{this.onCancel}} @isSaving={{this.save.isRunning}} />
+ *
+ * @param {boolean} isSaving - if true, disables the save and cancel buttons. useful if the onSubmit callback is a concurrency task
+ * @param {function} onCancel - called when form is canceled
+ * @param {function} onSubmit - called when form is saved, called with with the key value object containing patch data
+ * @param {string} submitError - error message string from parent if submit failed
+ */
+
+export default class KvPatchJsonForm extends Component {
+  @tracked jsonObject;
+  @tracked lintingErrors;
+
+  constructor() {
+    super(...arguments);
+    // prefill JSON editor with an empty object
+    this.jsonObject = JSON.stringify({ '': '' }, null, 2);
+  }
+
+  @action
+  handleJson(value, codemirror) {
+    codemirror.performLint();
+    this.lintingErrors = codemirror.state.lint.marked.length > 0;
+    if (!this.lintingErrors) {
+      this.jsonObject = value;
```

Comment:
We need to update `jsonObject` (or whatever we're passing into codemirror) even if it's not valid, because when we don't we get really weird jumpy behavior while typing. For this one, we could remove the check for linting errors on update and move it to submit instead, preventing submit when lint errors exist

This is also a bug in the KV experience that I'm working on fixing now.

Reply:
I'm reluctant to prevent submitting because still allow submitting when creating a new secret version. Plus, linting errors doesn't mean the data is incorrect it's just to enforce best practice, so I don't want to add more user friction and force them to use proper linting to submit the form. 

I can just remove the conditional so the value is updated every time. 🤔 But I'm wondering how the jumpiness occurred, I'm not seeing that locally.